README.md

```diff
@@ -605,13 +606,15 @@ Then, you will need to use the `e` option to manually choose which lines to add.
 <a href="unstaging-edits-and-staging-the-unstaged"></a>
 ### I want to stage my unstaged edits, and unstage my staged edits
 
-This is tricky. The best I figure is that you should stash your unstaged edits. Then, reset. After that, pop your stashed edits back, and add them.
+In many cases, You should unstage all files and then pick file you want and commit it.
+But assume that's what you want,You can create a temp commit to store your stage files, stage your unstaged files and stash them, Then, reset the temp commit and stash pop.
 
 ```sh
-$ git stash -k
-$ git reset --hard
-$ git stash pop
-$ git add -A
+$ git commit -m "WIP"
+$ git add . # also have untracked files
```

Comment:
Same for this. Let's assume that the user already has staged and unstaged edits.

Reply:
You can create a new local repo to try this.
I mark staged files m1 m2 unstaged files n1 n2 and untracked file n3.
[this link](https://stackoverflow.com/questions/20028507/git-stash-uncached-how-to-put-away-all-unstaged-changes/34681302#34681302) maybe help you.
I make a temp commit to store staged files.And then add unstaged files.
Actually, I suggest to handle it one by one...